native/src/main/java/io/ballerina/stdlib/http/transport/contractimpl/sender/http2/Http2ConnectionManager.java

```diff
@@ -147,6 +158,35 @@ void removeClientChannel(HttpRoute httpRoute, Http2ClientChannel http2ClientChan
         }
     }
 
+    void markClientChannelAsStale(HttpRoute httpRoute, Http2ClientChannel http2ClientChannel) {
+        Http2ChannelPool.PerRouteConnectionPool perRouteConnectionPool = fetchPerRoutePool(httpRoute);
+        if (perRouteConnectionPool != null) {
```

Comment:
When can this be null? Or did we add this as a safety check?

Reply:
When we remove the client channel, we have the same check. Hence added the check here as well